File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        area = []
        left, right = 0, len(height)-1

        while left < right :
            mh = height[left] if min(height[left], height[right]) == height[left] else height[right]
            area.append(mh*(right-left))

            if mh == height[left] :
                left += 1
            else :
                right -= 1

        return max(area)

File: 0011-container-with-most-water/0011-container-with-most-water.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0;
        int right = height.size()-1;
        int answer = -1;

        while (left<right) {
            int mh;
            mh = (height[left] < height[right]) ? height[left] : height[right];
            answer = max(answer, mh*(right-left));

            if (mh == height[left]) {
                left++;
            }
            else {
                right--;
            }
        }

        return answer;
    }
};